stl/inc/variant

```diff
@@ -1447,14 +1453,39 @@ _NODISCARD constexpr bool operator>=(const variant<_Types...>& _Left, const vari
     conjunction_v<is_nothrow_invocable_r<bool, greater_equal<>, const _Types&, const _Types&>...>) /* strengthened */ {
     // determine if _Left's index() is greater than _Right's, or equal and
     // _Left contains a value greater than or equal to _Right
-    using _Visitor             = _Variant_relop_visitor<greater_equal<>, _Types...>;
+    using _Visitor             = _Variant_relop_visitor<greater_equal<>, bool, _Types...>;
     const size_t _Left_offset  = _Left.index() + 1;
     const size_t _Right_offset = _Right.index() + 1;
     return _Left_offset > _Right_offset
         || (_Left_offset == _Right_offset
             && _Variant_raw_visit(_Right_offset - 1, _Right._Storage(), _Visitor{_Left._Storage()}));
 }
 
+#ifdef _HAS_CXX20
+struct _Variant_relop_3way_comparison {
+    template <class _Ty>
+    compare_three_way_result_t<_Ty> operator()(const _Ty& _Left, const _Ty& _Right) {
+        return _Left <=> _Right;
+    }
+};
+
+template <class... _Types>
+_NODISCARD constexpr common_comparison_category_t<compare_three_way_result_t<_Types>...>
+    operator<=>(const variant<_Types...>& _Left, const variant<_Types...>& _Right) noexcept(
+        conjunction_v<is_nothrow_invocable_r<common_comparison_category_t<compare_three_way_result_t<_Types>...>,
+            _Variant_relop_3way_comparison, const _Types&, const _Types&>...>) /* strengthened */ {
+    // determine the three-way comparison of _Left's and _Right's index, if equal
+    // return the three-way comparison of the contained value of _Left and _Right.

```

Comment:
```suggestion
    // Determine the three-way comparison of _Left's and _Right's index, if equal
    // return the three-way comparison of the contained values of _Left and _Right.
```

Reply:
The comment convention I'm enforcing here is "Either it's a complete sentence that begins with a capital letter and ends in punctuation, or it's _not_ a complete sentence doesn't begin with a capital letter and doesn't end with punctuation."

Reply:
I manage to reverse what I did and break the convention again (capital and no period). I will update again to conform with the function above in a final push, along with any additional feedback.